test/src/utils/kalman_multivariate.test.cpp

```diff
@@ -226,3 +362,131 @@ TEST_F(KalmanMathematics, handlesFilterWithControl)
       << expected_covariance_err;
   }
 }
+/**
+ * Struct used for testing the properties of the kalman filter under identity operations, those that
+ * do not modify neither the state vector nor the vocariance matrix on each filter().
+ */
+struct KalmanIdentity : public ::testing::Test {
+ protected:
+  unsigned int n = 2;
+  unsigned int m = 3;
+  unsigned int k = 1;
+
+  KalmanMultivariate kalman = KalmanMultivariate(n, m, k);
+  VectorXf x0 = VectorXf::Random(n);
+  VectorXf x1 = VectorXf::Random(n);
+  VectorXf z = VectorXf::Zero(m);
+  VectorXf u = VectorXf::Random(k);
+
+
+  MatrixXf P0 = MatrixXf::Random(n, n);
+  MatrixXf A = MatrixXf::Identity(n, n);
+  MatrixXf B = MatrixXf::Zero(n, k);
+  MatrixXf Q = MatrixXf::Zero(n, n);
+  MatrixXf H = MatrixXf::Zero(m, n);
+  MatrixXf R = MatrixXf::Identity(m, m);
+  MatrixXf P = MatrixXf::Random(n, n);
+  std::string zero_state_estimate_err = "Should handle zero vector as state estimate";
+  std::string arb_state_estimate_err = "Should handle any arbitrary vector as state estimate";
+  std::string zero_covariance_err = "Should handle zero state covariance";
+  std::string arb_covariance_err = "Should handle any arbitrary state covariance";
+
+  void SetUp()
+  {
+    // Assigns the matrices and vectors to the properties within the kalman object.
+    if (P.determinant() == 0) {
+       P = MatrixXf::Random(n, n);
+    }
+    kalman.setModels(A, B, Q, H, R);
+    kalman.setInitial(x0, P);
+  }
+  void TearDown() {}
+};
+
+/**
+ * Test fixture for testing whether the filter updates the X Vector and P Matrix
+ * accordingly, in this case the matrices have been created in a way that the state
+ * vector and the covariance should be mantained constant through all the filter() runs.
+ * Maths has been replicated throughout since to track the values updating and final
+ * assert at the end checks that the values are as expected.
+ * Checks that both the X vector and P Matrix are updated appropriately
+ */
+TEST_F(KalmanIdentity, handlesIdentity)
+{
+  for (int i = 0; i < 50; i++) {
```

Comment:
Apply the NUM_TESTDATA constant here aswell. Will need to create it again for the KalmanIdentity struct

Reply:
done